common/src/main/java/com/wynntils/overlays/minimap/MinimapOverlay.java

```diff
@@ -253,69 +256,69 @@ private void renderPois(
             cosRotationRadians = (float) -StrictMath.cos(rotationRadians);
         } else {
             sinRotationRadians = 0f;
-            cosRotationRadians = 0f;
+            cosRotationRadians = 1f;
         }
 
         float currentZoom = 1f / zoomRenderScale;
 
-        Stream<? extends Poi> poisToRender = Services.Poi.getServicePois();
-        poisToRender = Stream.concat(poisToRender, Services.Poi.getCombatPois());
-        poisToRender = Stream.concat(
-                poisToRender, Managers.Feature.getFeatureInstance(MainMapFeature.class).customPois.get().stream());
-        poisToRender = Stream.concat(poisToRender, Services.Poi.getProvidedCustomPois().stream());
-        poisToRender = Stream.concat(poisToRender, Models.Marker.getAllPois());
-        poisToRender = Stream.concat(
-                poisToRender, getMiniPlayerPois(renderRemotePartyPlayers.get(), renderRemoteFriendPlayers.get()));
+        // Get all MapData features as Pois
+        Stream<Pair<MapFeature, ResolvedMapAttributes>> mapFeatures = Services.MapData.getFeatures()
+                .filter(feature -> feature.isVisible(textureBoundingCircle))
+                .filter(feature -> feature instanceof TerritoryArea ? renderTerritories.get() : true)
+                .filter(feature -> !(feature instanceof SeaskipperDestinationArea))
+                .map(feature -> Pair.of(feature, Services.MapData.resolveMapAttributes(feature)))
+                .sorted(Comparator.comparing(pair -> pair.b().priority()));
 
-        MultiBufferSource.BufferSource bufferSource =
-                McUtils.mc().renderBuffers().bufferSource();
-
-        Poi[] pois = poisToRender.toArray(Poi[]::new);
-        for (Poi poi : pois) {
-            float dX = (poi.getLocation().getX() - (float) playerX) / zoomRenderScale;
-            float dZ = (poi.getLocation().getZ() - (float) playerZ) / zoomRenderScale;
-
-            if (followPlayerRotation.get()) {
-                float tempdX = dX * cosRotationRadians - dZ * sinRotationRadians;
-
-                dZ = dX * sinRotationRadians + dZ * cosRotationRadians;
-                dX = tempdX;
-            }
+        // FIXME: Add back the pois that are still not converted to MapData
```

Comment:
This FIXME is addressed now, isn't it?

Reply:
They'll be removed in their specific PRs. I think this list is maintained correctly, and there is a reason I've left it there. We'll come back to all FIXMEs in this PR before the final merge anyways.